lib/engine/step/g_18_ms/buy_train.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require_relative '../buy_train'
+
+module Engine
+  module Step
+    module G18MS
+      class BuyTrain < BuyTrain
+        def process_buy_train(action)
+          entity ||= action.entity
+          price = action.price
+          train = action.train
+          player = entity.player
+
+          president_assist, fee = @game.president_assisted_buy(entity, train, price)
+
+          if president_assist.positive?
+            player.spend(president_assist + fee, @game.bank)
+            @game.bank.spend(president_assist, entity)
+            @log << "#{player.name} pay #{@game.format_currency(president_assist)} to assist buy of #{train.name} train"
```

Comment:
Player pays $100 and an additional $5 fee to assist buying a 5 train.

Reply:
![image](https://user-images.githubusercontent.com/2631151/91896462-31549680-ec99-11ea-8cec-7878262854ed.png)

Reply:
6 trains normally cost $550. The log text are added before super method is called so it will be before the phase change message, and not connected to the log text about train buy. The emergency buy assist does appear together with the buy train log.
I think this might be good enough.